application-admintools-default/src/main/java/com/xwiki/admintools/internal/security/EntityRightsProvider.java

```diff
@@ -0,0 +1,235 @@
+
+    @Inject
+    private DocumentReferenceResolver<SolrDocument> solrDocumentReferenceResolver;
+
+    @Inject
+    @Named("secure")
+    private QueryManager secureQueryManager;
+
+    @Inject
+    private SolrUtils solrUtils;
+
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<String> documentReferenceResolver;
+
+    /**
+     * Retrieves a filtered and sorted list of {@link RightsResult} representing the rights for the given parameters.
+     *
+     * @param filters a map of filters to apply.
+     * @param sortColumn the column used for sorting.
+     * @param order the sorting order (asc or desc).
+     * @param entityType the type of entity for which rights are retrieved.
+     * @return a filtered and sorted {@link List} of {@link RightsResult}.
+     */
+    public List<RightsResult> getEntityRights(Map<String, String> filters, String sortColumn, String order,
+        String entityType)
+    {
+        try {
+            Set<RightsResult> rightsResults = new HashSet<>();
+            switch (filters.get(TYPE_KEY) == null ? "" : filters.get(TYPE_KEY)) {
+                case GLOBAL_TYPE:
+                    addGlobalRights(rightsResults, filters, entityType);
+                    break;
+                case SPACE_TYPE:
+                    addRights(rightsResults, filters, GLOBAL_RIGHTS_CLASS, SPACE_TYPE, entityType);
+                    break;
+                case PAGE_TYPE:
+                    addRights(rightsResults, filters, DOCUMENT_RIGHTS_CLASS, PAGE_TYPE, entityType);
+                    break;
+                default:
+                    addGlobalRights(rightsResults, filters, entityType);
+                    addRights(rightsResults, filters, GLOBAL_RIGHTS_CLASS, SPACE_TYPE, entityType);
+                    addRights(rightsResults, filters, DOCUMENT_RIGHTS_CLASS, PAGE_TYPE, entityType);
+                    break;
+            }
+
+            return applySort(rightsResults, sortColumn, order);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void addGlobalRights(Set<RightsResult> rightsResults, Map<String, String> filters, String entityType)
+        throws XWikiException
+    {
+        String wikiName = filters.get(WIKI_KEY);
+        String docName = filters.get(DOCUMENT_KEY);
+        String space = filters.get(SPACE_KEY);
+
+        DocumentReference docReference = resolveDocumentReference(wikiName);
+
+        boolean docNameMatches = isMatchingIgnoreCase(docName, docReference.getName());
+        boolean spaceMatches = isMatchingIgnoreCase(space, docReference.getLastSpaceReference().getName());
+
+        if (docNameMatches && spaceMatches) {
+            processDocumentRightsObjects(rightsResults, filters, GLOBAL_TYPE, docReference, GLOBAL_RIGHTS_CLASS,
+                entityType);
+        }
+    }
+
+    private DocumentReference resolveDocumentReference(String wikiName)
```

Comment:
Can you add a comment for this method? I'm not sure I understand, and the naming doesn't seem suited